### PR TITLE
enabling support for fully stateless client (fix for #220)

### PR DESCRIPTION
**Pull request by Dinesh2521**
As discussed in the issue [#220](https://github.com/singnet/snet-daemon/issues/220), state service has been modified to enable support for fully starless client.

### CODE REVIEW

**Comment by raamb** [comment]
@vsbogd would like your review on this. Thanks

File: escrow/escrow_test.go [hunk 1]
```diff
@@ -176,6 +176,8 @@ func (suite *PaymentChannelServiceSuite) channel() *PaymentChannelData {
 		Signer:           suite.signerAddress,
 		AuthorizedAmount: big.NewInt(0),
 		Signature:        nil,
+		PrevSignature:	  nil,
```
Comment (vsbogd) [inline escrow/escrow_test.go]: Please use non-trivial data to test copying code.

File: escrow/payment_channel_api.go [hunk 2]
```diff
@@ -203,6 +210,8 @@ var (
 	IncrementChannelNonce ChannelUpdate = func(channel *PaymentChannelData) {
 		channel.Nonce = (&big.Int{}).Add(channel.Nonce, big.NewInt(1))
 		channel.FullAmount = (&big.Int{}).Sub(channel.FullAmount, channel.AuthorizedAmount)
+		channel.PrevSignature = channel.Signature
```
Comment (vsbogd) [inline escrow/payment_channel_api.go]: Please add unit test to test this functionality

**Comment by Dinesh2521** [comment]
@vsbogd as you suggested all the unit testes have been added and renamed the new fields. please have a check and let me know.

File: escrow/escrow.go [hunk 3]
```diff
@@ -216,17 +216,19 @@ func (payment *paymentTransaction) Commit() error {
...
+			State:                payment.channel.State,
+			Sender:               payment.channel.Sender,
+			Recipient:            payment.channel.Recipient,
+			FullAmount:           payment.channel.FullAmount,
+			Expiration:           payment.channel.Expiration,
+			Signer:               payment.channel.Signer,
+			AuthorizedAmount:     payment.payment.Amount,
+			Signature:            payment.payment.Signature,
+			GroupID:              payment.channel.GroupID,
+			OldnonceSignedAmount: payment.channel.OldnonceSignedAmount,
```
Comment (vsbogd) [inline escrow/escrow.go]: Minor, but I would use `OldNonceSignetAmount` and `OldNonceSignature` as it looks a bit better.

File: escrow/state_service.proto [hunk 4]
```diff
@@ -42,6 +42,12 @@ message ChannelStateReply {
     bytes current_signed_amount = 2;
 
     // current_signature is a last signature sent by client with current_nonce
-    // it could be abset if none message was signed with current nonce
+    // it could be absent if none message was signed with current nonce
     bytes current_signature = 3;
+
+    // last amount which was signed by client with nonce=current_nonce - 1
+    bytes oldnonce_signed_amount = 4;
```
Comment (vsbogd) [inline escrow/state_service.proto]: Minor, but I would add additional undescope `old_nonce_signed_amount` and `old_nonce_signature`

File: escrow/payment_channel_storage_test.go [hunk 5]
```diff
@@ -56,18 +56,18 @@ func (suite *PaymentChannelStorageSuite) key(channelID int64) *PaymentChannelKey
...
+		Nonce:                big.NewInt(3),
+		Sender:               suite.senderAddress,
+		Recipient:            suite.recipientAddress,
+		GroupID:              [32]byte{123},
+		FullAmount:           big.NewInt(12345),
+		Expiration:           big.NewInt(100),
+		Signer:               suite.signerAddress,
+		AuthorizedAmount:     big.NewInt(0),
+		Signature:            nil,
+		OldnonceSignedAmount: big.NewInt(0),
```
Comment (vsbogd) [inline escrow/payment_channel_storage_test.go]: Please make amount and signature non-trivial.

**Comment by vsbogd** [comment]
@Dinesh2521 , sorry for confusing you with naming at the beginning but I think we need to slightly change variable names to make them look better (see comments).

**Comment by Dinesh2521** [comment]
@vsbogd updated the naming as you suggested. please have a check and let me know

**Comment by vsbogd** [comment]
Thanks!

**Pull request merged**